### PR TITLE
bugfix: adjusted CityScapes DataModule to proper torchvision api

### LINKED ISSUE #295: Cityscapes DataModule does not work due to wrong usage of torchvision.datasets.cityscapes.Cityscapes
## 🐛 Bug

CityScapes DataModule does not work at all. It seems to be related to wrong usage of the underlying `torchvision.datasets.cityscapes.Cityscapes` api.

### To Reproduce

Please see the code sample to reproduce. It throws an error: `TypeError: __init__() got an unexpected keyword argument 'train'`

#### Code sample

```python
from pl_bolts.datamodules.cityscapes_datamodule import CityscapesDataModule

dm = CityscapesDataModule('./data', num_workers=0, batch_size=1)
loader = dm.train_dataloader()
batch = next(iter(loader))
```

### Expected behavior

No crash and first batch returned.
### Environment

 - PyTorch Version: 1.6.0
 - OS: macOS
 - How you installed PyTorch: pip
 - Build command you used (if compiling from source): pip install `pip install https://github.com/PyTorchLightning/pytorch-lightning-bolts/archive/master.zip`
 - Python version: 3.8.5
 - CUDA/cuDNN version: irrelevant to this issue
 - GPU models and configuration: irrelevant to this issue
 - Any other relevant information:

### Additional context

I would be more than happy to submit a PR providing a fix for this issue.

### PR DESCRIPTION
**Pull request by pawelkobojek**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes #295 .

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/datamodules/cityscapes_datamodule.py [hunk 1]
```diff
@@ -177,3 +188,10 @@ def default_transforms(self):
             )
         ])
         return cityscapes_transforms
+
+    def default_target_transforms(self):
+        cityscapes_target_trasnforms = transform_lib.Compose([
+            transform_lib.ToTensor(),
+            transform_lib.Lambda(lambda t: t.squeeze())
```
Comment (pawelkobojek) [inline pl_bolts/datamodules/cityscapes_datamodule.py:195]: I have added this transform to labels as well as limited `target_type` to 'instance' and 'semantic' to ensure out-of-the-box operability with `pl_bolts.models.vision.SemSegment`. I'm wondering wheter it'd be more proper not to transform targets and return what torchvision's original CityScapes dataset returns and drop the requirements of instance/semantic target_type only. In such case, user of this DataModule would be responsible to transform targets to desired format but overall this DataModule would be more flexible

File: pl_bolts/datamodules/cityscapes_datamodule.py [hunk 2]
```diff
@@ -53,25 +53,29 @@ class CityscapesDataModule(LightningDataModule):
...
+            seed: int = 42,
             *args,
             **kwargs,
     ):
         """
         Args:
-            data_dir: where to save/load the data
-            val_split: how many of the training images to use for the validation split
+            data_dir: where to load the data from path, i.e. where directory leftImg8bit and gtFine or gtCoarse
+            are located
```
Comment (Borda) [inline pl_bolts/datamodules/cityscapes_datamodule.py:76]:
```suggestion
                are located
```

File: pl_bolts/datamodules/cityscapes_datamodule.py [hunk 3]
```diff
@@ -82,13 +86,18 @@ def __init__(
                 'You want to use CityScapes dataset loaded from `torchvision` which is not installed yet.'
             )
 
-        self.dims = (3, 32, 32)
+        if target_type not in ['instance', 'semantic']:
+            raise ValueError(f'Only "semantic" and "instance" target types are supported. Got {target_type}.')
+
+        self.dims = (3, 1024, 2048)
         self.DATASET = Cityscapes
```
Comment (Borda) [inline pl_bolts/datamodules/cityscapes_datamodule.py:93]: This shall be class constant

File: pl_bolts/datamodules/cityscapes_datamodule.py [hunk 4]
```diff
@@ -153,11 +161,19 @@ def val_dataloader(self):
...
     def test_dataloader(self):
         """
-        Cityscapes test set uses the test split
+        Cityscapes test set
         """
         transforms = self.default_transforms() if self.test_transforms is None else self.test_transforms
-
-        dataset = self.DATASET(self.data_dir, train=False, download=False, transform=transforms, **self.extra_args)
+        target_transforms = self.default_target_transforms(
+        ) if self.target_transforms is None else self.target_transforms
```
Comment (Borda) [inline pl_bolts/datamodules/cityscapes_datamodule.py:168]:
```suggestion
        target_transforms = self.default_target_transforms() if self.target_transforms is None else self.target_transforms
```
this is bad formatting

Comment (pawelkobojek) [reply pl_bolts/datamodules/cityscapes_datamodule.py]: Applying this suggestion would result in line too long error. In newer version I applied black autoformatting with `pyproject.toml` from parent `pytorch-lightning` repo.

Comment (Borda) [reply pl_bolts/datamodules/cityscapes_datamodule.py]: ok, pls see updated version

File: pl_bolts/datamodules/cityscapes_datamodule.py (hunk 3 above)
Comment (pawelkobojek) [reply pl_bolts/datamodules/cityscapes_datamodule.py]: Making it a class constant resulted in failing doctest (unwrap loop). I removed it and just use dataset class imported from torchvision, same as in other datamodules.

**Comment by TezRomacH** [comment]
Is there any progress on merging this PR? 🙂

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
Failing two tests are just SSL for specific version py3.7 and latest requirements
cc: @ananyahjha93 this is probably on master, mind check it... :]

**Pull request merged**